libs/autotagging-commons/src/main/java/org/opensearch/rule/service/IndexStoredRulePersistenceService.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rule.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ResourceNotFoundException;
+import org.opensearch.action.search.SearchRequestBuilder;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.autotagging.Attribute;
+import org.opensearch.autotagging.FeatureType;
+import org.opensearch.autotagging.Rule;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.rule.Utils.IndexStoredRuleParser;
+import org.opensearch.rule.Utils.IndexStoredRuleUtils;
+import org.opensearch.rule.action.GetRuleRequest;
+import org.opensearch.rule.action.GetRuleResponse;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.sort.SortOrder;
+import org.opensearch.transport.client.Client;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.autotagging.Rule._ID_STRING;
+
+/**
+ * This class encapsulates the logic to manage the lifecycle of rules at index level
+ * @opensearch.experimental
+ */
+public class IndexStoredRulePersistenceService implements RulePersistenceService {
+    /**
+     * The system index name used for storing rules
+     */
+    private final String indexName;
+    private final Client client;
+    private final FeatureType featureType;
+    private final int maxReturnSizeAllowedPerGetRequest;
+    private static final Logger logger = LogManager.getLogger(IndexStoredRulePersistenceService.class);
+    private static final Map<String, Object> indexSettings = Map.of("index.number_of_shards", 1, "index.auto_expand_replicas", "0-all");
+
+    @Inject
```

Comment:
I think this annotation is not required since we are managing the dependecies and initialization.